# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1302026980 -7200
#      Tue Apr 05 20:09:40 2011 +0200
# Node ID 677d14fe5d8e8431a9e00a17d98336d02626eed1
# Parent  5a6afd0e5213f74b9430a3fa5bee665b3e29c403
Errors are now passed through correctly, interactive mode enforced.

diff --git a/gui//client.cpp b/gui//client.cpp
--- a/gui//client.cpp
+++ b/gui//client.cpp
@@ -1,16 +1,17 @@
 #include "client.h"
 
 Client::Client(QString command)
     : m_command(command) {
     m_process.start(m_command, QProcess::ReadWrite);
-    connect(&m_process, SIGNAL(readyRead()), this, SLOT(reemitDataAvailable()));
+    connect(&m_process, SIGNAL(readyReadStandardOutput()), this, SLOT(reemitDataAvailable()));
     connect(&m_process, SIGNAL(stateChanged(QProcess::ProcessState)), this, SLOT(handleProcessStatusChange(QProcess::ProcessState)));
     connect(&m_process, SIGNAL(finished(int,QProcess::ExitStatus)), this, SLOT(handleProcessFinished(int,QProcess::ExitStatus)));
+    connect(&m_process, SIGNAL(readyReadStandardError()), this, SLOT(reemitErrorAvailable()));
 }
 
 void Client::send(QString content) {
     m_process.write(content.toLocal8Bit());
 }
 
 QString Client::fetch() {
     QString fetchedInput(m_process.readAllStandardOutput());
diff --git a/gui//mainwindow.cpp b/gui//mainwindow.cpp
--- a/gui//mainwindow.cpp
+++ b/gui//mainwindow.cpp
@@ -19,17 +19,17 @@ MainWindow::MainWindow(QWidget *parent)
 
 MainWindow::~MainWindow() {
 
 }
 
 void MainWindow::addOctaveTerminal() {
     OctaveTerminal *octaveTerminal = new OctaveTerminal;
     m_mdiArea->addSubWindow(octaveTerminal);
-    Client *octaveClient = ClientManager::clientManager().startProcess("octave");
+    Client *octaveClient = ClientManager::clientManager().startProcess("octave --interactive");
     octaveTerminal->assignClient(octaveClient);
 }
 
 void MainWindow::loadWebPage(QString title, QString url) {
 
     QWebView *webView = new QWebView();
     webView->setWindowTitle(title);
     webView->load(QUrl(url));
